### PR TITLE
RATEST-268: Refactoring TestData class and PatientGenerator class int…

### PR DESCRIPTION
**Pull request by Muta-Jonathan**
## Description of what I changed
Refactored code in PatientGenerator to TestData class

## Issue I worked on

see https://issues.openmrs.org/browse/RATEST-268

### CODE REVIEW

**Comment by Muta-Jonathan** [comment]
cc @dkayiwa @sherrif10 @kdaud

File: src/test/java/org/openmrs/contrib/qaframework/automation/RegistrationSteps.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 import static org.junit.Assert.assertNotNull;
 
 import org.openmrs.contrib.qaframework.RunTest;
-import org.openmrs.contrib.qaframework.helper.PatientGenerator;
+import org.openmrs.contrib.qaframework.helper.TestData;
 import org.openmrs.contrib.qaframework.helper.TestPatient;
```
Comment (sherrif10) [inline src/test/java/org/openmrs/contrib/qaframework/automation/RegistrationSteps.java:17]: @Muta-Jonathan  well done, Have you tested whether all the workflows are running perfect locally

Comment (Muta-Jonathan) [reply src/test/java/org/openmrs/contrib/qaframework/automation/RegistrationSteps.java:17]: have tried it but npm not running shows error!

Comment (kdaud) [reply src/test/java/org/openmrs/contrib/qaframework/automation/RegistrationSteps.java:17]: @Muta-Jonathan share the logs via pastebin.com

Comment (Muta-Jonathan) [reply src/test/java/org/openmrs/contrib/qaframework/automation/RegistrationSteps.java:17]: @kdaud [here](https://pastebin.com/pKTwdCuU) are the logs

Comment (Muta-Jonathan) [reply src/test/java/org/openmrs/contrib/qaframework/automation/RegistrationSteps.java:17]: this is just about my pc i think

**Comment by sherrif10** [comment]
I think you can copy them and share them in pastebin.com

**Comment by Muta-Jonathan** [comment]
https://pastebin.com/pKTwdCuU here @sherrif10

**Comment by kdaud** [comment]
@Muta-Jonathan share the output when you run;

`npm run refapp2Inpatient`

**Comment by Muta-Jonathan** [comment]
same issue https://pastebin.com/0yeYr1vb @kdaud

**Comment by kdaud** [comment]
which os are you running?

**Comment by Muta-Jonathan** [comment]
Windows 10

**Comment by kdaud** [comment]
@Muta-Jonathan what output do you get when you run the following commands
1. node -v 

2. npm -v

?

**Comment by Muta-Jonathan** [comment]
https://pastebin.com/PFVesark @kdaud

**Comment by kdaud** [comment]
@Muta-Jonathan seems like your missing module dependency for the test to run. What happens when you run

  `mvn clean install -DskipTests=true`

and then run 

  `npm run refapp2Inpatient`

?

**Comment by Muta-Jonathan** [comment]
https://pastebin.com/fgeKZfux .............. still the same .......... but at first with `mvn clean install -DskipTests=true` was **SUCCESS**

**Comment by kdaud** [comment]
update your npm tool by running `npm install -g npm@8.2.0` and share the feedback.

**Comment by Muta-Jonathan** [comment]
https://pastebin.com/iyVXYUdZ ....still de same 😢

**Comment by kdaud** [comment]
download a compatible driver engine( of the same version with the browser your running on your pc) for [firefox](https://github.com/mozilla/geckodriver/releases)/[chrome](https://chromedriver.chromium.org/downloads) for windows os. Then place the driver in the respective [folder](https://github.com/openmrs/openmrs-contrib-qaframework/tree/master/src/test/resources). 

Then run the test again!

**Comment by sherrif10** [comment]
Seems that also failed, @Muta-Jonathan  have you tried the one i had suggested of cleaning the m2.e folder and building again, maven is not building the dependencies cleary.

**Comment by kdaud** [comment]
> had suggested of cleaning the m2.e folder

This will results into loosing other dependencies that are innocent and will require building their respective projects again. Unless specifying the real folder where the `contrib` dependency are stored.

**Comment by sherrif10** [comment]
How about copy all the dependencies in m2.e  and pastethem in clean folder, then after delete them from m2e and do clean build, later you can easy return back your dependencies

**Comment by Muta-Jonathan** [comment]
tried it @sherrif10 but also failed

**Comment by kdaud** [comment]
@Muta-Jonathan any feedback on the comment [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/260#issuecomment-988648293)?

**Comment by Muta-Jonathan** [comment]
@kdaud  have tried it out both on firefox and chrome but same thing ..error

**Comment by kdaud** [comment]
@Muta-Jonathan you might have messed up with some files unknowingly, I would suggest to delete your local copy and clone a fresh one. And before introducing your changes ensure your able to run any of the existing tests !

**Comment by Muta-Jonathan** [comment]
@kdaud I clone a new git repo but still hadsame issues ...even without changes

**Comment by kdaud** [comment]
Add the windows configurations that are in pull/80  and share the outcome.

**Comment by Muta-Jonathan** [comment]
@kdaud am not getting u on this

**Comment by kdaud** [comment]
> am not getting u on this

checkout a pull request with id pull/80 containing windows configurations that have not been merged. Add those changes in your branch and run the test again!

**Comment by mherman22** [comment]
@Muta-Jonathan have you been running the tests before?
Then did, did your IDE throw errors in the pom.xml file and did you ignore those errors(using yourI DE) cuz if you did then thats the cause.

**Comment by Muta-Jonathan** [comment]
@mherman22  seen the talk thread ....how did u solve it coz it seems the same error on ma side 😃

**Comment by mherman22** [comment]
> 

well i took off to linux. 
However you can do as @kdaud said and follow [pull/80](https://github.com/openmrs/openmrs-contrib-qaframework/pull/80#issuecomment-901698855).

Otherwise i believe this exception has got to do with the `execution plugin`

**Comment by Muta-Jonathan** [comment]
I think switching mine will work coz after also changing as in pull 80 error still exists

**Comment by mherman22** [comment]
> I think switching mine will work coz after also changing as in pull 80 error still exists

its just a quick solution but the problem stays. if i had endured you would be having it so lets work to have [pull/80](https://github.com/openmrs/openmrs-contrib-qaframework/pull/80#issuecomment-901698855) merged, good enough you have the environment needed.

**Comment by kdaud** [comment]
@Muta-Jonathan share the setting residing in your `test.properties` file

**Comment by Muta-Jonathan** [comment]
```
webapp.url=https://qa-refapp.openmrs.org/openmrs
login.username=admin
login.password=Admin123
login.location=Pharmacy
webdriver=firefox
login.auto=false
headless=true
webdriver.gecko.driver=firefoxdriver/windows/geckodriver
```

## After chrome failing

**Comment by kdaud** [comment]
change the value of the login.auto to `true`

**Comment by Muta-Jonathan** [comment]
still failing

**Comment by kdaud** [comment]
are the logs constant?

**Comment by kdaud** [comment]
Share the outcome when you run `mvn clean install `against the module

**Comment by Muta-Jonathan** [comment]
not all errors are shown in the console https://pastebin.com/DF12TVYi

**Comment by kdaud** [comment]
> not all errors are shown in the console https://pastebin.com/DF12TVYi


share the full logs

**Comment by kdaud** [comment]
@Muta-Jonathan replace the configuration of windows in pom.xml file with;

for chrome at line `109-121`

```js
execution>
                        <id>make-windows-chromedriver-executable</id>
                        <phase>process-test-classes</phase>
                        <configuration>
                            <target>
                                <chmod file="target/test-classes/chromedriver/windows/chromedriver" perm="755" />
                            </target>
                        </configuration>
                        <goals>
                            <goal>run</goal>
                        </goals>
                    </execution>
```
and for firefox at line `148-160`
```js
<execution>
                        <id>make-windows-firefoxdriver-executable</id>
                        <phase>process-test-classes</phase>
                        <configuration>
                            <target>
                                <chmod file="target/test-classes/firefoxdriver/windows/geckodriver" perm="755"/>
                            </target>
                        </configuration>
                        <goals>
                            <goal>run</goal>
                        </goals>
                    </execution>
```

**Comment by kdaud** [comment]
ensure to remove the `.exe`  extension on the driver and then place it in the respective folder and run the test again

**Comment by kdaud** [comment]
@Muta-Jonathan any updates on the use-case [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/260#issuecomment-989152311)?

**Comment by Muta-Jonathan** [comment]
> @Muta-Jonathan any updates on the use-case [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/260#issuecomment-989152311)?

@kdaud due to failing tests on https://github.com/openmrs/openmrs-contrib-qaframework am unable to run the tests since I migrated to linux

**Comment by mherman22** [comment]
> > @Muta-Jonathan any updates on the use-case [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/260#issuecomment-989152311)?
> 
> @kdaud due to failing tests on https://github.com/openmrs/openmrs-contrib-qaframework am unable to run the tests since I migrated to linux

Those failing tests shouldn't deter you from running tests

**Comment by kdaud** [comment]
@muta did you take a look at the documentation that explains how to run the tests for refapp 2.x locally? https://wiki.openmrs.org/pages/viewpage.action?pageId=235277297

**Comment by kdaud** [comment]
worth taking a look and merge conflicting files and resolve.

**Comment by Muta-Jonathan** [comment]
@kdaud @mherman22 thanks got it I think the problem was with chrome driver so far since now it is running hope to finish soon

**Comment by kdaud** [comment]
@Muta-Jonathan some merge conflicts.

**Comment by Muta-Jonathan** [comment]
Opened a new PR here https://github.com/openmrs/openmrs-contrib-qaframework/pull/281

**Pull request closed without merging**